### ISSUE TITLE
Fail to build on Windows and Ubuntu 14.04 but builds on Mac OSX

### ISSUE THREAD

**Issue by raziel23x** [JesusFreke/smali#289]
When building on windows and Ubuntu I cannot get it to build it always failed to build with clean install of the OS

I am running VMware Workstation 11 to Run Clean installs of each OS on my windows machine and also running VMware Fusion Professional 7.1.1 on OSX to run test case

I ran the build instructions on
WINDOWS OS:
Windows 8.1 FAILED
Windows 8 FAILED
Windows 7 FAILED
Windows Vista FAILED
Window XP FAILED

Windows FAILED Test Results
https://dl.dropboxusercontent.com/u/12310125/APKTOOL_FAILED_TEST/JesusFreke_smali_tests_Failure.zip

Linux OS:
Ubuntu 12.04 FAILED
Ubuntu 14.04 FAILED
Ubuntu 15.04 FAILED

Mac OS X 10.10.3 Yosemite BUILD SUCCESSFUL

For some reason I cannot rap my head around it why it is Failing to build for me on windows and linux even when I am running Clean installs of the Operating Systems but It builds just fine on MAC OSX

**Comment by CalebFenton** [comment]
The test failed is org.jf.dexlib2.AccessorTest.testAccessors:
```junit.framework.AssertionFailedError: expected:<12> but was:<1>
	at junit.framework.Assert.fail(Assert.java:47)
	at junit.framework.Assert.failNotEquals(Assert.java:277)
	at junit.framework.Assert.assertEquals(Assert.java:64)
	at junit.framework.Assert.assertEquals(Assert.java:195)
	at junit.framework.Assert.assertEquals(Assert.java:201)
	at org.jf.dexlib2.AccessorTest.testAccessors(AccessorTest.java:125)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:44)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:180)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:41)
	at org.junit.runners.ParentRunner$1.evaluate(ParentRunner.java:173)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:220)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:86)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:49)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:64)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:50)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:360)
	at org.gradle.internal.concurrent.DefaultExecutorFactory$StoppableExecutorImpl$1.run(DefaultExecutorFactory.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)```

**Comment by Benjamin-Dobell** [comment]
I'm on Mac OS X 10.10.4 (14E11f) and encountering this...

**Comment by Benjamin-Dobell** [comment]
I'm really not quite sure what the test is trying to achieve...

It seems to be comparing the accessed value of ```Lorg/jf/dexlib2/AccessorTypes;->access$000(Lorg/jf/dexlib2/AccessorTypes;)Z``` against some predefined value (12) - an ```operationType```. However, I don't understand what ```operationType``` even represents, I can see that they're defined as:

```
    public static final int METHOD = 0;
    public static final int GETTER = 1;
    public static final int SETTER = 2;
    public static final int POSTFIX_INCREMENT = 3;
    public static final int PREFIX_INCREMENT = 4;
    public static final int POSTFIX_DECREMENT = 5;
    public static final int PREFIX_DECREMENT = 6;
    public static final int ADD_ASSIGNMENT = 7;
    public static final int SUB_ASSIGNMENT = 8;
    public static final int MUL_ASSIGNMENT = 9;
    public static final int DIV_ASSIGNMENT = 10;
    public static final int REM_ASSIGNMENT = 11;
    public static final int AND_ASSIGNMENT = 12;
    public static final int OR_ASSIGNMENT = 13;
    public static final int XOR_ASSIGNMENT = 14;
    public static final int SHL_ASSIGNMENT = 15;
    public static final int SHR_ASSIGNMENT = 16;
    public static final int USHR_ASSIGNMENT = 17;
```

But what that's got to do with the type of an accessed variable (in this case ```private boolean boolean_val```) is beyond me.

```SyntheticAccessorFSM.test()``` is indicating that the type of the accessed value is 1, which I'm guessing equates to a boolean value, because that's what it is. So why we're expecting a value of 12 seems very odd...

In summary - I don't understand what this test is trying to test.

**Comment by Benjamin-Dobell** [comment]
Alright... I finally think I understand what is going on:

```
        public void boolean_and(boolean val) {
            boolean_val &= val;
        }
```

We're expecting that to be compiled as a ```AND_ASSIGNMENT``` accessor *but* it's not - at least not with Java compiler on my machine. Instead it's being compiled using to accessors, a regular getter, and regular setter (i.e. ```GETTER``` and ```SETTER``` above).

This can be confirmed by using IntelliJ's decompiler on AccessorTypes.class, it looks like:

```
        public void boolean_and(boolean val) {
            AccessorTypes.this.boolean_val = AccessorTypes.this.boolean_val & val;
        }
```

i.e. No use of the JVM AND_ASSIGNMENT operator, just explicit getter, boolean and operator and an assignment.

So basically the problem is:

1. Whatever Java compiler I'm running sucks.
2. This test is invalid - because it would seem we can't assume that certain Java compilers won't completely suck.

**Comment by Benjamin-Dobell** [comment]
Just force pushed change:

```
Assert.assertEquals(operationType, accessedMemberType);
```
to:
```
Assert.assertEquals(accessedMemberType, operationType);
```

for consistency with the other Asserts just below it.

**Comment by JesusFreke** [comment]
Thanks for digging into this! Can someone mention what version of javac (e.g. javac -version) they're having this problem with?

I'm thinking the best way to handle this will be to check in the generated smali files and run the tests on them directly, and then have an auxillary gradle task that can be used to manually regenerate the smali files from the java source files, when needed.

**Comment by Benjamin-Dobell** [comment]
I'm not sure how to determine which version Gradle is picking - which is why I didn't mention it originally. However, my system javac is ```javac 1.8.0_05```. If you know how I can check which javac gradle is picking then please let me know.

**Comment by raziel23x** [comment]
I am running the same java 7 on windows linux and mac and mac is the only one who is successful with the building

**Comment by dhotrum** [comment]
Hi could somebody please help me? I have a Motorola droid mini (XT1030) It came from Verizon and I got it 2nd hand. Is there anyway to format the Cell and do a fresh install of Android like we do on a PC? I want to get rid of Verizon. Right now all my phone is good for is games and taking pics. What can I do?

**Comment by raziel23x** [comment]
https://github.com/dhotrum
no issue hiujacking make your own issue with this as your post has nothing to do with compiling the actual source to build the samli binaries

**Comment by JesusFreke** [comment]
I was able to reproduce this using sun's java 8, on ubuntu. This should be fixed by 0f1e9950
